nautobot/extras/api/views.py

```diff
@@ -77,12 +78,37 @@ def get_view_name(self):
         return "Extras"
 
 
+class NotesViewSetMixin(object):
+    @action(detail=True, url_path="notes", methods=["get", "post"])
+    def notes(self, request, pk=None):
+        """
+        API methods for returning or creating notes on an object.
+        """
+        obj = get_object_or_404(self.queryset, pk=pk)
+        if request.method == "POST":
+            content_type = ContentType.objects.get_for_model(obj)
+            data = request.data
+            data["assigned_object_id"] = obj.pk
+            data["assigned_object_type"] = f"{content_type.app_label}.{content_type.model}"
```

Comment:
Do we need a custom serializer class for this endpoint so that the OpenAPI schema / Swagger UI don't prompt/permit the user to explicitly specify these fields in the request data?

More generally, how does the Swagger UI look for this endpoint?

Reply:
I will need to look into this, Github is not showing me much context.